stubs/setuptools/setuptools/msvc.pyi

```diff
@@ -0,0 +1,169 @@
+import sys
+from typing import ClassVar, Final, TypedDict, overload, type_check_only
+from typing_extensions import LiteralString, NotRequired
+
+class winreg:
+    if sys.platform == "win32":
+        HKEY_USERS: Final[int]
+        HKEY_CURRENT_USER: Final[int]
+        HKEY_LOCAL_MACHINE: Final[int]
+        HKEY_CLASSES_ROOT: Final[int]
+    else:
+        HKEY_USERS: Final[None]
+        HKEY_CURRENT_USER: Final[None]
+        HKEY_LOCAL_MACHINE: Final[None]
+        HKEY_CLASSES_ROOT: Final[None]
+
+environ: dict[str, str]
+
+class PlatformInfo:
+    current_cpu: ClassVar[str]
+
+    arch: str
+
+    def __init__(self, arch: str) -> None: ...
+    @property
+    def target_cpu(self) -> str: ...
+    def target_is_x86(self) -> bool: ...
+    def current_is_x86(self) -> bool: ...
+    def current_dir(self, hidex86: bool = False, x64: bool = False) -> str: ...
+    def target_dir(self, hidex86: bool = False, x64: bool = False) -> str: ...
+    def cross_dir(self, forcex86: bool = False) -> str: ...
+
+class RegistryInfo:
+    if sys.platform == "win32":
+        HKEYS: Final[tuple[int, int, int, int]]
+    else:
+        HKEYS: Final[tuple[None, None, None, None]]
+
+    pi: PlatformInfo
+
+    def __init__(self, platform_info: PlatformInfo) -> None: ...
+    @property
+    def visualstudio(self) -> LiteralString: ...
+    @property
+    def sxs(self) -> LiteralString: ...
+    @property
+    def vc(self) -> LiteralString: ...
+    @property
+    def vs(self) -> LiteralString: ...
+    @property
+    def vc_for_python(self) -> LiteralString: ...
+    @property
+    def microsoft_sdk(self) -> LiteralString: ...
+    @property
+    def windows_sdk(self) -> LiteralString: ...
+    @property
+    def netfx_sdk(self) -> LiteralString: ...
+    @property
+    def windows_kits_roots(self) -> LiteralString: ...
+    @overload
+    def microsoft(self, key: LiteralString, x86: bool = False) -> LiteralString: ...
+    @overload
+    def microsoft(self, key: str, x86: bool = False) -> str: ...  # type: ignore[misc]
+    def lookup(self, key: str, name: str) -> str: ...
```

Comment:
These two can take a `StrPath` as the `key` param (and do represent relative paths, registry key paths to be exact).
But I'm pretty sure it's only meant to be a single final path segment, so relative `Path` support seem quite incidental. I think `str` is fine.